e2e/connect.test.tsx

```diff
@@ -0,0 +1,80 @@
+import '@ethersproject/providers'
+
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { act, render, RenderResult } from '@testing-library/react'
+import { tokens } from '@uniswap/default-token-list'
+
+import { SwapWidget } from '../src'
+
+const JSON_RPC_ENDPOINT = 'http://127.0.0.1:8545/'
+const provider = new JsonRpcProvider('http://127.0.0.1:8545/')
+
+class MockPromise<T = void> {
```

Comment:
why is a mock promise necessary? (I'm not familiar w/ the `act` fn below, forgive me if this should be obv

Reply:
I find this pattern makes sequencing more clear, instead of having to have the boilerplate of a Promise within the test.